tests/appstreamcli_validate.pm

```diff
@@ -0,0 +1,31 @@
+# vi: set shiftwidth=4 tabstop=4 expandtab:
+use base 'installedtest';
+use strict;
+use testapi;
+use utils;
+
+sub run {
+    my $self = shift;
+
+    # Start from the desktop
+    check_desktop_clean();
+
+    # Test that all the installed AppData files are valid.
+    # If you find that an application is installing an invalid AppData file,
+    # please fix the application *and* suggest that they add their own
+    # build-time `appstreamcli validate` test.
+
+    $self->user_console();
+    assert_script_run('for dir in $(echo $XDG_DATA_DIRS | tr ":" "\n"); do [[ $(compgen -G "${dir}/appdata/*.xml") ]] && appstreamcli validate ${dir}/appdata/*.xml; done');
```

Comment:
I went with `appstreamcli` because it’s in the images by default (hmm, should it be?) and `appstream-util` isn’t. If it’s possible to install `appstream-util` via flatpak then that would help, so I’ll give that a go. I suspect polkit might complicate the test, but perhaps I could install using `--user`.

I’ll give it a go later today and see what I can do.

Reply:
> it’s in the images by default (hmm, should it be?)

Good question. Looks like it's a runtime dependency of gnome-software. Curious1

Reply:
Blurgh, `appsteam-util validate` from flatpak doesn’t have permissions to read `/usr/share/appdata`. See [this failure](https://openqa.endlessm.com/tests/44491#step/appstreamcli_validate/10). I think it’ll be easiest to plough on with `appstreamcli` for now, rather than starting to faff around with copying files from multiple `appdata` directories into somewhere where the flatpakked `appstream-util` can read them. :(